include/nng/supplemental/util/nng_snprintf.h

```diff
@@ -0,0 +1,42 @@
+#ifndef NNG_SNPRINTF_H
+#define NNG_SNPRINTF_H
+
+#include <stdarg.h>  // NOLINT
+#include <stdio.h>   // NOLINT
+
+#if !defined(__cplusplus) && defined(_MSC_VER)
+#define inline __inline
+#endif  //__cplusplus && _MSC_VER
+
+#ifdef __cplusplus
+extern "C" {
+#endif  //__cplusplus
+
+#if !defined(_TRUNCATE)
+#define _TRUNCATE ((size_t)-1)
+#endif  //_TRUNCATE
+
+inline int nng_snprintf(char* const _Buffer, int const _BufferCount, char const* const _Format, ...) {
+  int ret = 0;
+  va_list args;
+  va_start(args, _Format);
+#if _MSC_VER
+  ret = vsnprintf_s(_Buffer, _BufferCount, _TRUNCATE, _Format, args);
```

Comment:
I see -- this does the right thing to terminate safely.

I still am not thrilled with doing this work to support older MSVC -- there are three newer major releases of MSVC that would work without needing (2015, 2017, and 2019).

Exposing this functionality here in this way isn't right either.  If we want to make it a public function (and I'm not sure we do -- I don't intend NNG to become the universal platform portability library), we should stick a prototype in supplemental/util/platform.h and put MSVC implementation specific details in the win32 port layer.  Ideally it would look at the MSC version and only provide a builtin wrapper around _vsnprintf_s if the version was too old.

Reply:
I understand that due to resource constraints and over all sanity, projects have a need to limit it's support for older platforms. However, some projects that depend on `nng` have limitations on what compiler they can use. For one example, WinCE uses a hybrid compiler between VS 2012 and VS 2013. That compiler version will never be updated, so thats what we have to work with. Supporting VS13 is a step towards supporting WinCE.

In an ideal world, everyone would be using the latest and greatest, however, sometimes that's not possible. :/

Going forward, I like they suggestion of moving this in to the platform layer instead.